core/src/main/java/com/linecorp/armeria/client/DefaultHttpClient.java

```diff
@@ -42,22 +45,38 @@ public HttpResponse execute(HttpRequest req) {
     }
 
     private HttpResponse execute(@Nullable EventLoop eventLoop, HttpRequest req) {
-        final String originalPath = req.path();
-        final String newPath = concatPaths(uri().getRawPath(), originalPath);
-        final PathAndQuery pathAndQuery = PathAndQuery.parse(newPath);
-        if (pathAndQuery == null) {
-            req.abort();
-            return HttpResponse.ofFailure(new IllegalArgumentException("invalid path: " + newPath));
-        }
-
-        final HttpRequest newReq;
-        if (newPath != originalPath) {
-            newReq = HttpRequest.of(req, req.headers().toBuilder().path(newPath).build());
+        if (uri() == UNDEFINED_URI) {
```

Comment:
I was thinking instead of making some clients absolute and some not, we could examine the request instead, if it is an absolute url use it otherwise treat it as always. It has the big advantage of making it easier to handle Location header in redirects which I've found to be hard right now. Having to configure two clients in the same way, one to handle relative the other absolute is not so intuitive.

Would it be possible to do that?

Reply:
@anuraaga Thanks for fast feedback.  I understood why it matters in retrying client. I will remove this condition and replace it with checking request header path. 
I assume if request header path is absolute URL then it would be higher priority than client level URI. 
Check it by adding test case for redirect. 😀